src/apps/create-patron-user-info/CreatePatron.entry.tsx

```diff
@@ -54,15 +56,17 @@ export interface CreatePatronProps
     CreatePatronTextProps {}
 
 const CreatePatronEntry: FC<CreatePatronProps> = () => {
-  const userToken = hasToken("user") ? getToken("user") : null;
+  const u = useUrls();
+  const dashboardUrl = u("dashboardUrl");
 
-  // The application using this app should handle the case where the user is not logged in.
-  // Eg by returning 403 Forbidden or redirecting to the login page.
-  if (!userToken) {
-    return null;
-  }
+  useEffect(() => {
+    // If the user has already been registered, redirect to the dashboard.
+    if (!isUnregistered()) {
+      redirectTo(dashboardUrl);
+    }
+  }, [dashboardUrl]);
 
-  return <CreatePatron />;
+  return isUnregistered() ? <CreatePatron /> : null;
```

Comment:
Do we need this if statement here if we otherwise already redirect the user to dashboard?

Reply:
This is a very good question.
The thing is that useEffect is activated after rendering. So if the user is NOT unregistered we would get a glimpse of weirdness and even errors if we retun the `CreatePatron` component.
One could ask if we need the useEffect at all. But it works like this and I would rather test that part later. I'll leave a TODO about that.